src/graph/LookupExecutor.cpp

```diff
@@ -126,15 +130,23 @@ Status LookupExecutor::prepareYield() {
                 return Status::SyntaxError("Not supported yet");
             }
             auto* prop = col->expr();
-            if (prop->kind() != Expression::kAliasProp) {
-                return Status::SyntaxError("Expressions other than AliasProp are not supported");
+            const AliasPropertyExpression* aExpr;
+            if (prop->kind() == Expression::kAliasProp) {
+                aExpr = dynamic_cast<const AliasPropertyExpression*>(prop);
+            } else if (isEdge_ && prop->kind() == Expression::kEdgeSrcId) {
+                aExpr = dynamic_cast<const EdgeSrcIdExpression*>(prop);
+            } else if (isEdge_ && prop->kind() == Expression::kEdgeDstId) {
+                aExpr = dynamic_cast<const EdgeDstIdExpression*>(prop);
+            } else {
+                return Status::SyntaxError("Expressions other than "
+                                           "AliasProp/kEdgeSrcId/kEdgeDstId are not supported");
             }
-            auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(prop);
             auto st = checkAliasProperty(aExpr);
             if (!st.ok()) {
                 return st;
             }
-            if (schema->getFieldIndex(aExpr->prop()->c_str()) < 0) {
+            if (prop->kind() == Expression::kAliasProp &&
+                schema->getFieldIndex(aExpr->prop()->c_str()) < 0) {
```

Comment:
So there's some confusion here by the logic above

Reply:
The logic here is to judge whether the current attribute column is valid only when the value of `prop->kind() == Expression::kAliasProp` is true. In other words, when `isEdge_ && prop->kind() == Expression::kAliasProp` or `isEdge_ && prop->kind() == Expression::kEdgeDstId` is true  , there is no need to go through the logic here